### DISCUSSION TITLE
reading a vtu mesh file

### DISCUSSION THREAD

**Discussion by dealmeidavf** [idaholab/moose#17598] [Q&A Meshing] [answered]
Hello,
Is there an example of how to read a `vtu` mesh file while identifying the boundaries so boundary conditions can be applied.
I can see the input example [here](test/tests/mesh/named_entities/name_on_the_fly.i)

```
[Mesh]
  file = three_block.e

  # These names will be applied on the fly to the
  # mesh so that they can be used in the input file
  # In addition they will show up in the output file
  block_id = '1 2 3'
  block_name = 'wood steel copper'

  boundary_id = '1 2'
  boundary_name = 'left right'
[]
```
How do I get or create the `block_id`and `boundary_id` labels in a `vtu` file? If anyone would have a small, say 2D, `vtu` ASCII file available to share, I would appreciate.
Thanks,

Valmor

**Comment by ke7kto** [comment, marked as answer]
Try opening your .vtu file in paraview. You should be able to find your block and subdomain ids there, and those will be the same IDs MOOSE will use. If you have problems, you can run your MOOSE app with the `--mesh-only` flag, and output an exodus file that will definitely have the information you're looking for, which you can inspect in paraview.

**Reply by jakpatel** [reply]
Hi there, I'm facing similar issues. I'm trying to do a diffusion solve on a .vtu mesh. I am not able to run it. How should I proceed? Any help is very highly appreciated!

Thanks,
J

**Comment by dealmeidavf** [comment]
The `VTK` files consist of metadata `vtm` and data `vtu`. In the `vtm` file:
```
<VTKFile type="vtkMultiBlockDataSet" version="1.0" byte_order="LittleEndian" header_type="UInt64">
  <vtkMultiBlockDataSet>
    <Block index="0" name="Element Blocks">
      <DataSet index="0" name="Unnamed block ID: 1" file="brick.vtu"/>
    </Block>
    <Block index="4" name="Side Sets">
      <DataSet index="0" name="inflow"/>
      <DataSet index="1" name="outflow"/>
      <DataSet index="2" name="duct"/>
    </Block>
  </vtkMultiBlockDataSet>
  <FieldData>
    <DataArray type="Float64" Name="TimeValue" NumberOfTuples="1" format="ascii" RangeMin="0" RangeMax="0">
      0
    </DataArray>
  </FieldData>
</VTKFile>
```
I see the side sets. The actual data is in the `brick.vtu` file. If I run `MOOSE` with `--mesh-only` it produces an `.e` output with no side sets. Also, I am unable to read the `vtm`/`vtu` data directly with `MOOSE`. It wants a file with extension `vtu`. This is probably why the `.e` file does not have the side sets when I give `MOOSE` the `.vtu` file (no metadata). Is there a way to read xml `VTK` files in `MOOSE` and be able to access the side sets so I can enforce boundary conditions?
Thanks.

**Comment by WilkAndy** [comment]
I, too, am interested in this.  I've never sidesets to work, but have never tried too hard.  We have a lot of scripts that generate vtk files, and i end up converting to exodus and spending plenty of time defining sidesets.  If anyone has a recipe to follow regarding MOOSE + VTK, i'd love to hear it.

**Reply by GiudGiud** [reply]
dont think vtu is supported but vtk is 
https://mooseframework.inl.gov/source/meshgenerators/FileMeshGenerator.html

meshio is a great one for converting between formats btw, and cubit can do a few conversions too
